resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/ExecutorKubernetesCredentialsFeatureStep.scala

```diff
@@ -17,19 +17,23 @@
 package org.apache.spark.deploy.k8s.features
 
 import org.apache.spark.deploy.k8s.{KubernetesConf, SparkPod}
-import org.apache.spark.deploy.k8s.Config.KUBERNETES_SERVICE_ACCOUNT_NAME
+import org.apache.spark.deploy.k8s.Config.{KUBERNETES_DRIVER_SERVICE_ACCOUNT_NAME, KUBERNETES_EXECUTOR_SERVICE_ACCOUNT_NAME}
 import org.apache.spark.deploy.k8s.KubernetesUtils.buildPodWithServiceAccount
 
 private[spark] class ExecutorKubernetesCredentialsFeatureStep(kubernetesConf: KubernetesConf)
   extends KubernetesFeatureConfigStep {
-  private lazy val driverServiceAccount = kubernetesConf.get(KUBERNETES_SERVICE_ACCOUNT_NAME)
+
+  private lazy val driverServiceAccount = kubernetesConf.get(KUBERNETES_DRIVER_SERVICE_ACCOUNT_NAME)
+  private lazy val executorServiceAccount =
+    kubernetesConf.get(KUBERNETES_EXECUTOR_SERVICE_ACCOUNT_NAME)
 
   override def configurePod(pod: SparkPod): SparkPod = {
       pod.copy(
-        // if not setup by the pod template fallback to the driver's sa,
-        // last option is the default sa.
+        // if not setup by the pod template, fallback to the executor's sa,
+        // if executor's sa is not setup, the last option is driver's sa.
         pod = if (Option(pod.pod.getSpec.getServiceAccount).isEmpty) {
-          buildPodWithServiceAccount(driverServiceAccount, pod).getOrElse(pod.pod)
```

Comment:
Hi, @liyinan926 . The current behavior is fallback to `driverServiceAccount` first, isn't it?

Reply:
Oh, right. Then we should stick to the current behavior.

Reply:
hmm. I think you're right by reading the code. 
However I am pretty sure that in vanilla Spark if you set the driver service account name, the executors will be on default. 
I think possibly BasicExecutorFeatureStep ignores the config? I think an integration test will help here.

Reply:
If then, could you file a JIRA for that, @ecapoccia ? It sounds like worth to have a test case at least.

Reply:
Actually I don't think it makes much sense to default to the driver's service account given that the executors don't manage pods/services or any other K8S resources. Defaulting to the `default` service account should be the right behavior.

Reply:
Yep. I agree. If the current behavior is wrong, we can fix that in a separate JIRA and PR since the default behavior was orthogonal.

Reply:
@liyinan926 agree with you. they are very different, and should have different identities.
Also -- apologies in advance I am betraying the fact that I'm old here..:)
It violates the principle of least astonishment https://en.wikipedia.org/wiki/Principle_of_least_astonishment
If one sets the driver service account to XYZ and says nothing about the executor service account, I don't think that one expects the executor's service account to be XYZ.

Reply:
Again, you are mentioning the existing behavior before this PR. Please file a JIRA. Otherwise, nothing will happen here for that. It's a way of open source. 😄 
> If one sets the driver service account to XYZ and says nothing about the executor service account, I don't think that one expects the executor's service account to be XYZ.